exporters/metrics/src/main/java/com/google/cloud/opentelemetry/metric/MetricTranslator.java

```diff
@@ -104,28 +114,69 @@ static MetricDescriptor mapMetricDescriptor(
         .forEach((key, value) -> builder.addLabels(mapAttribute(key, value)));
 
     MetricDataType metricType = metric.getType();
-    if (GAUGE_TYPES.contains(metricType)) {
-      builder.setMetricKind(MetricDescriptor.MetricKind.GAUGE);
-    } else if (CUMULATIVE_TYPES.contains(metricType)) {
-      builder.setMetricKind(MetricDescriptor.MetricKind.CUMULATIVE);
-    } else {
-      logger.error(
-          "Metric type {} not supported. Only gauge and cumulative types are supported.",
-          metricType);
-      return null;
-    }
-    if (LONG_TYPES.contains(metricType)) {
-      builder.setValueType(MetricDescriptor.ValueType.INT64);
-    } else if (DOUBLE_TYPES.contains(metricType)) {
-      builder.setValueType(MetricDescriptor.ValueType.DOUBLE);
-    } else {
-      logger.error(
-          "Metric type {} not supported. Only long and double types are supported.", metricType);
-      return null;
+    switch (metricType) {
+      case LONG_GAUGE:
+        builder.setMetricKind(MetricDescriptor.MetricKind.GAUGE);
+        builder.setValueType(MetricDescriptor.ValueType.INT64);
+        break;
+      case DOUBLE_GAUGE:
+        builder.setMetricKind(MetricDescriptor.MetricKind.GAUGE);
+        builder.setValueType(MetricDescriptor.ValueType.DOUBLE);
+        break;
+      case LONG_SUM:
+        builder.setValueType(MetricDescriptor.ValueType.INT64);
+        fillSumType(metric.getLongSumData(), builder);
+        break;
+      case DOUBLE_SUM:
+        builder.setValueType(MetricDescriptor.ValueType.DOUBLE);
+        fillSumType(metric.getDoubleSumData(), builder);
+        break;
+      case HISTOGRAM:
+        fillHistogramType(metric.getDoubleHistogramData(), builder);
+        break;
+      default:
+        logger.error(
+            "Metric type {} not supported. Only gauge and cumulative types are supported.",
+            metricType);
+        return null;
     }
     return builder.build();
   }
 
+  private static void fillHistogramType(
+      DoubleHistogramData histogram, MetricDescriptor.Builder builder) {
+    builder.setValueType(MetricDescriptor.ValueType.DISTRIBUTION);
+    switch (histogram.getAggregationTemporality()) {
+      case DELTA:
+        builder.setMetricKind(MetricDescriptor.MetricKind.DELTA);
+        return;
+      case CUMULATIVE:
+        builder.setMetricKind(MetricDescriptor.MetricKind.CUMULATIVE);
+        return;
+      default:
+        logger.error(
+            "Histogram type {} not supported. Only delta and cumulative types are supported.",
+            histogram);
+        return;
+    }
+  }
+
+  private static void fillSumType(SumData<?> sum, MetricDescriptor.Builder builder) {
+    // TODO: Treat non-monotonic sums as gauges?
```

Comment:
Hmm looking at examples of up down counter from the spec:

> Example uses for UpDownCounter:
> 
> - the number of active requests
> - the number of items in a queue?

Sound like Gauges to me, but you lose the semantic that these things can be aggregated with Sum() function. How would these work as CUMULATIVE in cloud

Reply:
Modified these to be gauges for now.